### PR TITLE
Added AutoDiscovery Docs

### LINKED ISSUE #303: Add Docs Section for AutoDiscovery
The AutoDiscovery is currently not present at our docs page except for the https://www.securecodebox.io/docs/how-tos/autodiscovery how-to page.

The auto-discovery should probably have its own subsection in the sidebar.

Topics to address include:

- Installation
- Gradual rollout (namespace by namespace in existing clusters)
- Page explaining Service AutoDiscovery and how to configure it
- Page explaining Container AutoDiscovery and how to configure it

### PR DESCRIPTION
**Pull request by Ilyesbdlala**
closes #303

### CODE REVIEW

**Review by snoopy-cat — COMMENTED** [review]
Nice work 🙂 I got it running with the Service-AutoDiscovery Guide 🥳 
I have some small comments, mostly typos.

File: docs/auto-discovery/service-auto-discovery.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+We also have to enable Service AutoDiscovery. This is done through the `config.serviceAutoDiscovery.enabled` parameter in the helm chart. It goes as follows:
+
+```bash
+helm upgrade --namespace securecodebox-system --install auto-discovery-kubernetes secureCodeBox/auto-discovery-kubernetes --set config.serviceAutoDiscovery.enabled=true
+```
+### Rollout
+AutoDiscovery has three modes of operation:
+* enabled-per-namespace (default) : scans every container in an enabled namespace
+* enabled-per-resource :  scans every enabled container
+* all (scans every container in the whole cluster!)
```
[the remark this answers is not in the document]
Comment (J12934) [reply docs/auto-discovery/service-auto-discovery.md]: Ah perfect, yes this should not really be here as the inclusionModes are configured once and not on the Service / Container AutoDiscovery level. I've moved it to the general AutoDiscovery Install Page

File: docs/auto-discovery/service-auto-discovery.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+title: "Service AutoDiscovery"
+sidebar_label: "Service AutoDiscovery"
+path: "docs/auto-discovery/service-auto-discovery"
+sidebar_position: 2
+---
+
+The Service AutoDiscovery will create a scheduled scan with the given parameters (see [readme](https://github.com/secureCodeBox/secureCodeBox/blob/main/auto-discovery/kubernetes/README.md) for config options) for each Kubernetes service it detects. (It is possible to scan APIs that require authentication, see the [ZAP Advanced](../scanners/zap-advanced.md) documentation).
```
[the remark this answers is not in the document]
Comment (J12934) [reply docs/auto-discovery/service-auto-discovery.md]: I added a bit of text describing this somewhat cryptic sentance a bit more :)

[the remark this answers is not in the document]
Comment (snoopy-cat) [reply docs/auto-discovery/service-auto-discovery.md]: Great, thank you, that's a lot better now 👍

File: docs/auto-discovery/service-auto-discovery.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (snoopy-cat) [reply docs/auto-discovery/service-auto-discovery.md]: Ah nice :) Then it makes sense that we also cut it from the Container Page (`docs/auto-discovery/container-auto-discovery.md`), doesn't it?

**Review by snoopy-cat — COMMENTED** [review]
Thanks for the changes! Found one last typo :)

File: docs/auto-discovery/service-auto-discovery.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (J12934) [reply docs/auto-discovery/service-auto-discovery.md]: ahhh yes, i somehow thought it was just here... 🤦

**Review by snoopy-cat — APPROVED** [review]
👍 🥳

**Comment by Weltraumschaf** [comment]
@Ilyesbdlala can you rebase please?

**Pull request merged**